tests/unit/src/test/scala/tests/DefinitionLspSuite.scala

```diff
@@ -74,8 +74,8 @@ object DefinitionLspSuite extends BaseLspSuite("definition") {
            |package a
            |import java.util.concurrent.Future/*Future.java*/ // unused
            |import scala.util.Failure/*Try.scala*/ // unused
-           |object Main/*MainSuite.scala:6*/ extends App/*App.scala*/ {
```

Comment:
I hadn't realized we were calling "find references" in this test suite. That explains why the test was flaky, "find references" triggers a background cascade compilation. Generally, any discarded `Future[T]` is a likely source of flaky test failures. It's a good change to only test for definitions in `DefinitionLspSuite` 👍